app/vmagent/remotewrite/remotewrite.go

```diff
@@ -587,6 +587,9 @@ func newRemoteWriteCtx(argIdx int, at *auth.Token, remoteWriteURL *url.URL, maxI
 }
 
 func (rwctx *remoteWriteCtx) MustStop() {
+	sas := rwctx.sas.Swap(nil)
```

Comment:
nit: would be nice to explain order in comments, so future readers could understand dependency between steps